src/js/hca/config/form.js

```diff
@@ -361,11 +362,17 @@ const formConfig = {
             // TODO: this should really be a dateRange, but that requires a backend schema change. For now
             // leaving them as dates, but should change these to get the proper dateRange validation
             lastEntryDate: currentOrPastDateUI('Start of service period'),
-            lastDischargeDate: currentOrPastDateUI('Date of discharge'),
+            lastDischargeDate: {
+              'ui:title': 'Date of discharge',
+              'ui:widget': 'date'
+            },
             dischargeType: {
               'ui:title': 'Character of discharge',
+              // TODO: Use a constant instead of a magic string
+              'ui:required': (formData) => !moment(_.get('lastDischargeDate', formData), 'YYYY-MM-DD').isAfter(moment().startOf('day')),
```

Comment:
Is it necessary to include anything in .isAfter() for a future date? We use an alternative syntax in src/js/common/schemaform/SaveInProgressIntro.jsx on line 67.

Reply:
I'm not sure. [The docs](https://momentjs.com/docs/#/query/is-after/) say that omitting the argument defaults to the current time, but what if the date is earlier in the same day? I think this is the right way to do it, but it's been so long ago I forget the exact reasons why.

Basically the argument in the `.isAfter()` just ensures that we're working with the beginning of the day instead of an arbitrary time in the middle of the day. Like I said, I don't remember why that was important, though.